conda/misc.py

```diff
@@ -228,6 +228,12 @@ def clone_env(prefix1, prefix2, verbose=True, quiet=False, index_args=None):
     dists = r.dependency_sort({d.quad[0]: d for d in urls.keys()})
     urls = [urls[d] for d in dists]
 
+    precs = tuple(index[dist] for dist in dists)
+    disallowed = tuple(MatchSpec(s) for s in context.disallowed_packages)
+    for prec in precs:
```

Comment:
Why is this code duplicated here and https://github.com/conda/conda/pull/6932/files#diff-315c016a32ef55653e67795ff2b18b7cR468

Reply:
The code here is for the "explicit" code path, which addresses #5802.  This code path is ancient and in desperate need for some reworking, which will probably happen whenever someone gets to #6147.

Reply:
Sounds good :-)